Update access graph version and add Docker deployment warning

jakule opened the pull request:
The access graph version in the config.json file has been updated from 1.20.1 to 1.20.4. Also, a warning notice about Docker deployment suitability for testing and development purposes only has been added in the Docker deployment page.
Included instruction to make the tls keys readable after the default docker user has been changed.

jakule committed:
Update access graph version and add Docker deployment warning

The access graph version in the config.json file has been updated from 1.20.1 to 1.20.4. Also, a warning notice about Docker deployment suitability for testing and development purposes only has been added in the Docker deployment page.

jakule committed:
Update access control documentation for non-root containers

The Access Graph service documentation has been updated to reflect that containers now run as non-root users by default starting from version 1.20.4. Instructions have been amended to guide users on the necessary steps to ensure certificate files are readable by the new non-root user. Obsolete file permission tweaks were also removed.

justinas commented on docs/pages/access-controls/access-graph/self-hosted.mdx:
````diff
@@ -29,8 +29,19 @@ to Teleport Enterprise customers.
+  - Starting from version 1.20.4 of the Access Graph service, the container by default runs as a non-root user.
+    Make sure the certificate files are readable by the user running the container. You can set correct permissions with the following command:
+    ```console
+    $ sudo chown 65532 /path/to/tls.key
````
Nit: we use `/etc/access_graph/tls.key` as the path in the example config further below, so maybe use that here as well.

jakule replied:
Changed

justinas commented on docs/pages/access-controls/access-graph/self-hosted.mdx:
````diff
@@ -43,7 +54,7 @@ The host CA can be retrieved and saved into a file in one of the following ways:
 $ sudo mkdir /etc/access_graph
 $ curl -s 'https://<Var name="teleport.example.com" />/webapi/auth/export?type=tls-host' | sudo tee /etc/access_graph/teleport_host_ca.pem
-```
+    ```
````
unintended change?

justinas commented on docs/pages/access-controls/access-graph/self-hosted.mdx:
```diff
@@ -29,8 +29,19 @@ to Teleport Enterprise customers.
 - A TLS certificate for the Access Graph service
   - The TLS certificate must be issued for "server authentication" key usage,
     and must list the IP or DNS name of the TAG service in an X.509 v3 `subjectAltName` extension.
+  - Starting from version 1.20.4 of the Access Graph service, the container by default runs as a non-root user.
```
nit: "the container runs as a non-root user by default." might be a nicer phrasing?

jakule committed:
Update Access Graph documentation

The documentation for the Access-Graph service has been updated to clarify the information about non-root user and adjust the command-line path. It also includes corrections to some formatting issues for a better readability.

jakule commented:
Friendly ping @r0mant @zmb3 @ptgott

ptgott commented on docs/pages/access-controls/access-graph/self-hosted.mdx:
````diff
@@ -29,8 +29,19 @@ to Teleport Enterprise customers.
     and must list the IP or DNS name of the TAG service in an X.509 v3 `subjectAltName` extension.
+  - Starting from version 1.20.4 of the Access Graph service, the container runs as a non-root user by default.
+    Make sure the certificate files are readable by the user running the container. You can set correct permissions with the following command:
+    ```console
````
```suggestion
    ```code
```
For consistency with the rest of the docs

ptgott commented on docs/pages/access-controls/access-graph/self-hosted.mdx:
```diff
@@ -29,8 +29,19 @@ to Teleport Enterprise customers.
 
+<Notice type="warning">
+    The deployment with Docker is suitable for testing and development purposes. For production deployments,
+    consider using a container orchestration system like Kubernetes and Helm.
+    Refer to [Helm chart for Access Graph](self-hosted-helm.mdx) for instructions.
```
```suggestion
    consider using the Teleport Access Graph Helm chart to deploy this service on Kubernetes.
    Refer to [Helm chart for Access Graph](self-hosted-helm.mdx) for instructions.
```

Suggested to be more direct about the fact that we maintain a Helm chart for deploying TAG.

jakule committed:
Apply suggestions from code review

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>
